### ISSUE TITLE
[Feature Request] Function that checks if a resource name already exists

### ISSUE THREAD

**Issue by kevin-lindsay-1** [tilt-dev/tilt#3809]
For example:
```
def special_resource_definition(...):
  if not resource_exists('helper'):
    local_resource('helper', ...)
```

**Comment by landism** [comment]
Could you elaborate on your use case a bit?

While the proposed solution seems like it'd be a no-brainer, it doesn't really fit into the Tiltfile execution model:
1. The Tiltfile executes, loading yaml and specifying options.
2. Tilt takes all the info specified during Tiltfile execution (e.g., yaml, docker builds, and options) and assembles it into resources.

Which is to say, no resources exist while the Tiltfile is executing. While that'd be simple to change in the case of local_resource, things are rather more complicated for k8s_resource.

Maybe we can come up with another solution for your problem.

**Comment by kevin-lindsay-1** [comment]
My current use case will actually be solved by #3571, however I can pretty easily come up with cases where it would be very convenient to have. I can likely always work around the lack, but this would likely solve for those.

For example,
I use nested tiltfiles combined with `meta`, which allows me to spin up slices of architecture. I don't normally run databases out of kube, but I do via helm when I'm using tilt.

- Tiltfile (Gateway/Master `metarepo`)
  - Tiltfile for `User` service
    - Tiltfile at `util/database.Tiltfile` -> k8s_resource('database')
  - Tiltfile for `Cart` service
    - Tiltfile at `util/database.Tiltfile` -> k8s_resource('database')

Hopefully that's a decent enough example.
